aggregator/src/aggregator.rs

```diff
@@ -226,21 +227,30 @@ impl<
 
 #[cfg(test)]
 mod tests {
-    use substrate_test_runtime::Hash as THash;
-
-    use crate::aggregation::aggregator::{AggregatorError, BlockSignatureAggregator};
+    use crate::{
+        aggregator::{AggregatorError, BlockSignatureAggregator},
+        multicast::MockHash,
+        Metrics,
+    };
 
     type TestMultisignature = usize;
     const TEST_SIGNATURE: TestMultisignature = 42;
 
-    fn build_aggregator() -> BlockSignatureAggregator<THash, TestMultisignature> {
+    struct MockMetrics;
+    impl Metrics<MockHash> for MockMetrics {
+        fn report_aggregation_complete(&mut self, _h: MockHash) {
+            todo!()
```

Comment:
Then do. :>

Reply:
Unless you don't need it because of the `Option`,, in which case plx delete.

Reply:
Suggestion: remove that `Option`, provide some default empty trait implementation and use it instead.

Reply:
left with empty implementation

Reply:
Huh, I just noticed that you could provide a blanked implementation for `Option<T>` when `T: Metrics` that does nothing on `None` and marginally simplify the interface by requiring `M: Metrics` instead of `Option<M>`. But that is just something to think about, I'm not even sure it's actually reasonable.